plugins/tiddlywiki/tiddlyweb/sse-server.js

```diff
@@ -3,92 +3,70 @@ title: $:/plugins/tiddlywiki/tiddlyweb/sse-server.js
+        var jsonChanges = JSON.stringify(changes);
+        eventServer.emitEvent("wiki-change", "change", jsonChanges, filter);
+    });
+    wikis.push(wiki);
 }
 
 /*
 Setup this particular wiki if we haven't seen it before
 */
-function ensureWikiSetup(wiki) {
-	if(wikis.indexOf(wiki) === -1) {
-		setupWiki(wiki);
-	}
-}
-
-/*
-Return the array of connections for a particular wiki
-*/
-function getWikiConnections(wiki) {
-	return connections[wikis.indexOf(wiki)];
-}
-
-function addWikiConnection(wiki,connection) {
-	getWikiConnections(wiki).push(connection);
-}
-
-function removeWikiConnection(wiki,connection) {
-	var wikiConnections = getWikiConnections(wiki);
-	var index = wikiConnections.indexOf(connection);
-	if(index !== -1) {
-		wikiConnections.splice(index,1);
-	}
+function ensureChannelSetup(channel, wiki) {
+    // setup wikis for the wiki-change channel
+    if (channel === "wiki-change" && wikis.indexOf(wiki) === -1) setupWiki(wiki);
 }
 
-function handleConnection(request,state,emit,end) {
-	if(isDisabled(state)) {
-		return;
-	}
+var eventServer = new Journal();
 
-	ensureWikiSetup(state.wiki);
-	// Add the connection to the list of connections for this wiki
-	var connection = {
-		request: request,
-		state: state,
-		emit: emit,
-		end: end
-	};
-	addWikiConnection(state.wiki,connection);
-	request.on("close",function() {
-		removeWikiConnection(state.wiki,connection);
-	});
+// this filter is called for the emitter route, which recieves 
+// messages from clients and forwards them to all listeners. It 
+// does not affect messages sent directly by the server. 
+// We don't use it in tiddlyweb so just set it to false
+eventServer.emitterFilter = function (sender) {
+    // do not allow clients to broadcast
+    // they can't anyway unless a route is specified
+    return function () { return false; };
 }
 
-function isDisabled(state) {
-	return state.server.get("sse-enabled") !== "yes";
-}
-
-// Import the ServerSentEvents class
-var ServerSentEvents = require("$:/core/modules/server/server-sent-events.js").ServerSentEvents;
-// Instantiate the class
-var events = new ServerSentEvents("plugins/tiddlywiki/tiddlyweb", handleConnection);
-// Export the route definition for this server sent events instance
-module.exports = events.getExports();
-
-})();
+// Export the route definition for this server sent events handler. 
+// We don't need an emitter route, otherwise we could put the common 
+// instance in a library tiddler export and require it in both files.
+module.exports = eventServer.handlerExports(
+    "plugins/tiddlywiki/tiddlyweb",
+    function (request, response, state) {
+        if (state.server.get("tiddlyweb-sse-enabled") !== "yes"
+            || state.params[0] !== "wiki-change"
+        ) {
+            response.writeHead(404);
+            response.end();
+            return;
+        }
+        ensureChannelSetup(state.params[0], state.wiki);
+        eventServer.handler(request, response, state);
```

Comment:
there should be no spaces between the parameters in function calls eg: `eventServer.handler(request,response,state);` ... That's valid for _all_ function calls